src/main/resources/reference.conf

```diff
@@ -123,6 +175,230 @@ resourceTypes = {
     authDomainConstrainable = true
     reuseIds = false
   }
+  controlled-user-shared-workspace-resource = {
```

Comment:
Is there any reason to apply auth domains to these resources?

Reply:
I don't know, I didn't include them here because I wasn't sure. @davidangb or @dvoet, do we want controlled resources to be auth domain constrainable?

Reply:
I think that when/if Rawls (or current Terra as a whole) wants to use controlled resources, they will need to support auth domains. Else, we can't properly constrain access to the resource.

Reply:
Sounds good, added this to the four types of controlled resources

Reply:
I'm now second-guessing my comment, and suggest getting an authoritative answer from Cloud Integration. My concern here is making sure that the cloud ACLs on the resource itself in Google or Azure or wherever are correct, and that if a user creates a resource inside an auth-domained workspace, cloud-native access to the resource is prevented for anyone not in the auth domain.  I don't know if this requires the Sam resource to support  auth domains or not;  I *assume* it does but don't know for sure.

Reply:
From CI:
> Synchronization with google groups gets extra complicated when there are auth domains involved so we are not allowing this for now. This remains a problem to figure out

at the moment, that means we can't add controlled resources to workspaces with auth domains. I don't know if the current plan is to replace auth domains with policy enforcement or fix this restriction in Sam, but either way this is a blocker on migrating existing Rawls workspaces to WSM. 
Given that, my inclination for now is to disallow auth domains on resources until we know we need them. I'm still pretty shaky on this whole area, though